subsys/bluetooth/services/ots/otc.c

```diff
@@ -169,15 +126,91 @@ static int read_attr(struct bt_conn *conn,
 		    struct bt_otc_internal_instance_t *inst,
 		    uint16_t handle, bt_gatt_read_func_t cb);
 
-static void print_oacp_response(enum bt_ots_oacp_proc_type req_opcode,
-				enum bt_ots_oacp_res_code result_code)
+/* L2CAP callbacks */
+static void tx_done (struct bt_gatt_ots_l2cap *l2cap_ctx,
+		     struct bt_conn *conn)
+{
+	/* Not doing any writes yet */
+	BT_ERR("Unexpected call, context: %p, conn: %p", l2cap_ctx, conn);
+}
+
+static ssize_t rx_done (struct bt_gatt_ots_l2cap *l2cap_ctx,
+			 struct bt_conn *conn, struct net_buf *buf)
+{
+	uint32_t offset = cur_inst->rcvd_size;
+	bool is_complete = false;
+	struct bt_otc_obj_metadata *cur_object =
+		&cur_inst->otc_inst->cur_object;
+	int cb_ret;
+
+	BT_DBG("Incoming L2CAP data, context: %p, conn: %p, len: %u, offset: %u",
+	       l2cap_ctx, conn, buf->len, offset);
+
+	cur_inst->rcvd_size += buf->len;
+
+	if (cur_inst->rcvd_size >= cur_object->current_size) {
+		is_complete = true;
+	}
+
+	if (cur_inst->rcvd_size > cur_object->current_size) {
+		BT_WARN("Received %u but expected maximum %u",
+			cur_inst->rcvd_size, cur_object->current_size);
+	}
+
+
+	cb_ret = cur_inst->otc_inst->cb->content_cb(conn, offset,
+						    buf->len, buf->data,
+						    is_complete, 0);
+
+	if (is_complete) {
+		uint32_t rcv_size = cur_object->current_size;
+		int err;
+
+		BT_DBG("Received the whole object (%u bytes). "
+		       "Disconnecting L2CAP CoC", rcv_size);
+		err = bt_gatt_ots_l2cap_disconnect(l2cap_ctx);
+
+		if (err < 0) {
```

Comment:
```suggestion
		err = bt_gatt_ots_l2cap_disconnect(l2cap_ctx);
		if (err < 0) {
```

Reply:
Is this some guideline?
Personally, I often prefer a blank line here, I feel it improves readability.
(We have a guideline to have a blank line after a right curly brace - logically it seems to make sense to have one at the start of the block as well.)

Reply:
It is not - But it is something that @jhedberg have often requested from me :) I've added an entry for this in the BT subsystem guidelines discussion: https://github.com/zephyrproject-rtos/zephyr/discussions/41120#discussioncomment-2066740

Reply:
Changed